service/src/main/java/bio/terra/externalcreds/services/OAuth2Service.java

```diff
@@ -136,4 +147,36 @@ public OAuth2User getUserInfo(ClientRegistration providerClient, OAuth2AccessTok
     var userRequest = new OAuth2UserRequest(providerClient, accessToken);
     return new DefaultOAuth2UserService().loadUser(userRequest);
   }
+
+  /**
+   * Revoke the refresh token associated with the linked account. This is a best attempt and will
+   * not throw an exception if it fails.
+   *
+   * @param providerProperties provider properties
+   * @param linkedAccount linked account
+   */
+  public void revokeRefreshToken(
+      ProviderProperties providerProperties, LinkedAccount linkedAccount) {
+    String revokeEndpoint = providerProperties.getRevokeEndpoint();
+
+    // spring security does not seem to have a revocation client so use nimbus
+    var req =
+        new TokenRevocationRequest(
+            URI.create(revokeEndpoint),
+            new ClientSecretBasic(
+                new ClientID(providerProperties.getClientId()),
+                new Secret(providerProperties.getClientSecret())),
+            new RefreshToken(linkedAccount.getRefreshToken()));
+    try {
+      var response = req.toHTTPRequest().send();
+      log.info(
+          "Token revocation request for user [{}], provider [{}] returned status [{}] with the result: [{}]",
+          linkedAccount.getUserId(),
+          linkedAccount.getProvider().toString(),
+          response.getStatusCode(),
+          response.getContent());
+    } catch (IOException e) {
+      log.error("revoke refresh token failure", e);
+    }
```

Comment:
is there anything sensitive in the successful response content or stack trace we don't want to log?

Reply:
good thought, but I don't think so